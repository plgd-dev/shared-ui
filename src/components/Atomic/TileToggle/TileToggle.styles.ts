import { css } from '@emotion/react'
import { colors } from '../_utils/colors'
import { fontPrimary } from '../_utils/commonStyles'
import { ThemeType, get } from '../_theme'

export const tileToggle = (theme: ThemeType) => css`
    background: ${get(theme, `TileToggle.background`)};
    border-radius: 8px;
    padding: 24px;
    display: flex;
    justify-content: space-between;
    align-items: center;
`

export const name = css`
    font-family: ${fontPrimary};
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    line-height: 16px;
    color: ${colors.primaryDarken};
    display: flex;
    align-items: center;
`

export const loading = css`
    margin-left: 8px;
`

export const headline = css`
    font-family: 'Poppins', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 18px;
    margin: 0 0 4px 0;
    color: ${colors.neutral500};
`

export const content = css`
    font-family: 'Poppins', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 22px;
    color: ${colors.neutral800};
`

export const tags = css`
    margin: -4px;

    .tag {
        margin: 4px;
    }
`
