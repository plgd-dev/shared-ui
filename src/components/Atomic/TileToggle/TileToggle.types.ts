export type Props = {
    checked: boolean
    className?: string
    darkBg?: boolean
    name: string
    onChange: (e: any) => void
    loading?: boolean
}
