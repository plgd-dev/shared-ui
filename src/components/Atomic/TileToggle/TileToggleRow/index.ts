export { default } from './TileToggleRow'
export * from './TileToggleRow'
