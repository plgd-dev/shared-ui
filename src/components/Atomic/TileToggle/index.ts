export { default } from './TileToggle'
export * from './TileToggle'

export { default as TileRow } from './TileToggleRow'
export * from './TileToggleRow'
