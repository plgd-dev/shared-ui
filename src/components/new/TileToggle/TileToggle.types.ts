import { ReactNode } from 'react'

export type Props = {
    checked: boolean
    className?: string
    name: string
    onChange: (e: any) => void
    loading?: boolean
}
