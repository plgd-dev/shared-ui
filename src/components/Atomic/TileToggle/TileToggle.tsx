import React, { FC, memo } from 'react'
import { ClipLoader } from 'react-spinners'

import { Props } from './TileToggle.types'
import * as styles from './TileToggle.styles'
import Switch from '../Switch'
import { colors } from '../_utils/colors'

const TileToggle: FC<Props> = memo((props) => {
    const { className, darkBg, dataTestId, name, checked, onChange, loading } = props
    return (
        <div className={className} css={[styles.tileToggle, darkBg && styles.darkBg]} data-test-id={dataTestId}>
            <div css={styles.name}>
                {name}
                {loading && <ClipLoader color={colors.primaryDarken} css={styles.loading} size={16} />}
            </div>
            <Switch checked={checked} dataTestId={dataTestId?.concat('-switch')} disabled={loading} loading={loading} onChange={onChange} />
        </div>
    )
})

TileToggle.displayName = 'TileToggle'

export default TileToggle
