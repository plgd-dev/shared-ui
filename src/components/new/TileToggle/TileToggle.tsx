import React, { FC, memo } from 'react'
import { Props } from './TileToggle.types'
import * as styles from './TileToggle.styles'
import Switch from '../Switch'
import { ClipLoader } from 'react-spinners'
import { colors } from '../_utils/colors'

const TileToggle: FC<Props> = memo((props) => {
    const { className, name, checked, onChange, loading } = props
    return (
        <div className={className} css={styles.tileToggle}>
            <div css={styles.name}>
                {name}
                {loading && <ClipLoader color={colors.primaryDarken} css={styles.loading} size={16} />}
            </div>
            <Switch checked={checked} disabled={loading} loading={loading} onChange={onChange} />
        </div>
    )
})

TileToggle.displayName = 'TileToggle'

export default TileToggle
