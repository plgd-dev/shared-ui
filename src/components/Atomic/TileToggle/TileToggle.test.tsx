import React from 'react'
import { render } from '@testing-library/react'
import TileToggle from './TileToggle'

describe('<TileToggle>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <TileToggle checked={true} name='Checked' onChange={() => {}} />
                <TileToggle checked={false} name='UnChecked' onChange={() => {}} />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
