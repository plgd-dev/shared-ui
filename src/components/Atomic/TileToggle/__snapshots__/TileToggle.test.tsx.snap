// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TileToggle> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: #F6F7F9;
  border-radius: 8px;
  padding: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 600;
  font-size: 14px;
  line-height: 16px;
  color: #0A2965;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  width: 32px;
  height: 20px;
  border-radius: 26.6667px;
  background: #D7D8DA;
  position: relative;
}

.emotion-4 {
  opacity: 0;
  width: 0;
  height: 0;
}

.emotion-4:checked~.emotion-6 {
  background: #2261AE;
}

.emotion-4:checked~.emotion-6:before {
  -webkit-transform: translateX(12px);
  -moz-transform: translateX(12px);
  -ms-transform: translateX(12px);
  transform: translateX(12px);
}

.emotion-5 {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #ccc;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  width: 32px;
  height: 20px;
  border-radius: 26.6667px;
}

.emotion-5:before {
  position: absolute;
  content: '';
  left: 2px;
  bottom: 2px;
  background-color: white;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 50%;
}

.emotion-5:before {
  width: 16px;
  height: 16px;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        Checked
      </div>
      <label
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <input
            checked=""
            class="emotion-4"
            type="checkbox"
          />
          <span
            class="emotion-5 emotion-6"
          />
        </div>
      </label>
    </div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        UnChecked
      </div>
      <label
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <input
            class="emotion-4"
            type="checkbox"
          />
          <span
            class="emotion-5 emotion-6"
          />
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
